R/layer_templates.R

```diff
@@ -0,0 +1,321 @@
+#' Create, view, extract, remove, and use Tplyr layer templates
+#'
+#' There are several scenarios where a layer template may be useful. Some
+#' tables, like demographics tables, may have many layers that will all
+#' essentially look the same. Categorical variables will have the same count
+#' layer settings, and continuous variables will have the same desc layer
+#' settings. A template allows a user to build those settings once per layer,
+#' then reference the template when the Tplyr table is actually built.
+#'
+#' This suite of functions allows a user to create and use layer templates.
+#' Layer templates allow a user to pre-build and reuse an entire layer
+#' configuration, from the layer constructor down to all modifying functions.
+#' Furthermore, users can specify parameters they may want to be
+#' interchangeable. Additionally, layer templates are extensible, so a template
+#' can be use and then further extended with additional layer modifying
+#' functions.
+#'
+#' Layers are created using `new_layer_template()`. To use a layer, use the
+#' function `use_template()` in place of `group_count|desc|shift()`. If you want
+#' to view a specific template, use `get_layer_template()`. If you want to view
+#' all templates, use view_layer_templates. And to remove a layer template use
```

Comment:
Probably want back ticks and parenthesis for view_layer_templates to match other functions.

Reply:
> As I'm not very familiar with the use case or the internals, my review is more focused on design and documentation vs. implementation. Some additional comments below.
> 
> I really like this idea! I can definitely see how it could help standardization and reduce redundant code.
> 
> Things I recommend
> 
> * Run `devtools::document` again. Some of the docs need updating along with NAMESPACE.
> * Add layer templates to the table in the Tplyr Options vignette.
> 
> Things for consideration
> 
> * As a user the syntax for viewing a single template (`get_layer_template`) vs. all templates (`view_layer_templates`) could be confusing. In particular 'get' vs. 'view'. View makes more sense to me given the language of the documentation but get is clear as well. I'd think about picking one and use it for both single and all templates.

So if I had to use one I would use `get`. Is your suggestion to have `get_layer_templates()` instead of `view_`? So add the s and have both `get_layer_template()` and `get_layer_templates()`

Reply:
Fixed

Reply:
For view/get, yes I suggest adding the 's' to have `get_layer_template` and `get_layer_templates`

Reply:
@asbates  I'm actually going to keep tplyr.layer_templates _out_ of the options vignette because I intentionally don't want people to touch the option.